source/common/router/upstream_request.cc

```diff
@@ -750,82 +651,10 @@ void UpstreamRequest::onPoolReady(std::unique_ptr<GenericUpstream>&& upstream,
 
   for (auto* callback : upstream_callbacks_) {
     callback->onUpstreamConnectionEstablished();
-    return;
-  }
-
-  auto* headers = parent_.downstreamHeaders();
```

Comment:
I'm probably missing something, but seems that the removed code from this point won't be called if there are no upstream-callbacks. I guess that if the tests pass, then this case is handled correctly.

Reply:
it won't be called if there were upstream callbacks - the return I removed above took care of that.